rest/rest-resources/src/test/java/org/apache/brooklyn/rest/resources/CatalogResourceTest.java

```diff
@@ -487,6 +495,79 @@ public void testAddMissingItem() {
         //equivalent to HTTP response 404 text/html
         addAddCatalogItemWithInvalidBundleUrl("classpath://missing-jar-file.txt");
     }
+    
+    @Test
+    public void testOsgiBundleWithBom() throws Exception {
+        TestResourceUnavailableException.throwIfResourceUnavailable(getClass(), OsgiStandaloneTest.BROOKLYN_TEST_OSGI_ENTITIES_PATH);
```

Comment:
I worry that this test might only pass because this OSGi bundle's symbolic name starts with `org.apache.brooklyn...`. It is therefore white-listed. If we fail to find the resource `classpath:/org/apache/brooklyn/test/osgi/entities/icon.gif`, we'll look inside all the installed bundles whose symbolic names match the white-listed regexes.

If you changed this to use `BROOKLYN_TEST_OSGI_ENTITIES_COM_EXAMPLE_PATH`, does it still pass?

If so, how? How does it know that the item in the catalog came from this given bundle, when it has no `brooklyn.libraries` section?! Do we need to auto-add the `brooklyn.libraries` section when we are explicitly adding the new item to the catalog?